src/main/scala/is/hail/HailContext.scala

```diff
@@ -497,18 +497,59 @@ class HailContext private(val sc: SparkContext,
       hadoopConf.set("io.compression.codecs",
         codecs.replaceAllLiterally("org.apache.hadoop.io.compress.GzipCodec", "is.hail.io.compress.BGzipCodecGZ"))
 
-    val vds = LoadVCF(this,
-      header,
-      inputs,
-      storeGQ,
-      nPartitions,
-      sitesOnly,
-      ppAsPL,
-      skipBadAD)
+    val settings = VCFSettings(storeGQ, sitesOnly, ppAsPL, skipBadAD)
+    val reader = new GenotypeRecordReader(settings)
+    val result = LoadVCF(this, reader, header, inputs, nPartitions, sitesOnly)
 
     hadoopConf.set("io.compression.codecs", codecs)
 
-    vds
+    VariantSampleMatrix(this, VariantMetadata(result.sampleIds,
```

Comment:
This block looks identical to the one below in importVCFsGeneric.  If pull this into LoadVCF, it can return a VSM[T] instead of a LoadVCFResults (which can be deleted).